Python/ceval.c

```diff
@@ -4744,8 +4744,10 @@ PyEval_EvalCodeEx(PyObject *_co, PyObject *globals, PyObject *locals,
         Py_DECREF(defaults);
         return NULL;
     }
+#ifndef NDEBUG
     PyCodeObject *code = (PyCodeObject *)_co;
     assert ((code->co_flags & (CO_NEWLOCALS | CO_OPTIMIZED)) == 0);
+#endif
```

Comment:
You could just inline the case instead.

Reply:
I've updated the PR to use inline casting, I will merge this PR after CI is passed 
cc @vstinner